### ISSUE TITLE
Sony Bravia tv - no surround / 5.1 sound from games

### ISSUE THREAD

**Issue by sdestan** [moonlight-stream/moonlight-android#482]
Hi ,

everything is fine but my games run in stereo only.
I've checked 5.1 enabling from settings on moonlight on my sony bravia tv.

how to play games in 5.1 ?

thanks

**Comment by cgutman** [comment]
What games have you tried?

**Comment by sdestan** [comment]
most of the games
witcher 3 
tomb raider
resident evil 7

**Comment by sdestan** [comment]
I've been researching this case for hours.

My receiver is connected through ARC to my tv.
When I start moonlight and run steam games , receiver always show STEREO. ( auto detection is on ).
When I start other apps like netflix , video, etc. receiver ( same connection through arc ) shows dts , dolby etc.

need your help please , anyone who uses moonlight on their tv and getting 5.1 sound from their steam games ?

thanks

**Comment by vrokolos** [comment]
ARC and optical requires dolby digital encoded 5.1 sound. There is not enough bandwidth for 5.1 uncompressed sound. This means that it requires realtime DD encoding which is not currently provided by nvidia. Let's all go bug nvidia to support it!

**Comment by Xiller** [comment]
I have a fireTV, if someone read this:
1080p - 60fps - 10Mbit (not 20!) - And for flickering fix: use only one Monitor ;).

Okay, the last thing or two are:
5.1 Audio and Xbox wireless controller blinking.

@vrokolos 
Could you give me or us a Link in the Nvidia Forum where i can lovefull spam the right thread? : ) @5.1 problem

**Comment by vrokolos** [comment]
Start one :) They probably need licensing from dolby as well i think

**Comment by vrokolos** [comment]
Yes the need to add support for Dolby Live encoding. This is not free however

**Comment by 90Ninety** [comment]
> Yes they need to add support for Dolby Live encoding. This is not free however

Hmm , its a strange one , even when connecting locally ( PC to TV) the audio is encoded to stereo / 2 Channel .  Years ago , I can remember that my old GTX 260 had a SPDIF passthough ( physical connector )  , which could take an audio stream from the motherboard  and pass it through the DVI  
I cannot understand why , years later NVIDIA  have ommited this and wont allow a proper 5.1 true DD/DDL encoding over HDMI . A standard PS3 does this , XBOX also ( I think ) even my £30 dvd player can encode DD through HDMI . 

Would the video cards themselves need licencing , if they just supported a passthough , rather than the encoding ( for example  allowing an input from a audio card , then outputting this audio on a few of the unused HDMI PIns ) 

Even if the there was a dammed expensive encoding option - I would pay good money but , NVIDIA or AMD do not offer this . They could simply add a module which uses the SLI header , or release a special card variants which allow true Dolby Digital encoding , or even a software which require a purchased key . It is simply unacceptable to not support DD or DTS which have been the standard for many many years .

**Comment by vrokolos** [comment]
No neither of those do dolby live encoding. Your cheap DVD player passes the already encoded DD stream to the receiver. It doesn't encode RAW audio to DD on the fly. PS3 and XBOX don't do live encoding either. It's the game developers / engine that has to do it! Anyway HDMI supports up to 7.1 RAW audio which also means everything else as well (THX / DD etc). HDMI using ARC only has enough bandwidth to support 2 channel raw audio or up to 5.1 encoded. It can't do 5.1 raw so it needs to encode / compress it on the fly.

**Comment by 90Ninety** [comment]
PS3 and XBOX don't do live encoding either. It's the game developers / engine that has to do it!
I think I meant encoding * 
So the audio stream is encoded into the source , Then why do I have to buy a separate audio card ( Asus Xonar) to send a decode DDL signal to the 5.1 ?    From what I have experienced all encoded  games support DD / DDL when decoded by my Xonar hardware  . In my eyes its an issue with  manufacturers rather than the game . I am really struggling to understand why the audio cannot be passed though the HDMI

**Comment by vrokolos** [comment]
Trying to ELI5:
Encoded DD 5.1 is something like mp3
Decoded DD 5.1 is something like flac

Your hardware DD decoder converts mp3 to flac so that speakers can play it.
HDMI supports flac 5.1
HDMI ARC doesn't support flac 5.1 cause of bandwidth. It can only do mp3 5.1

Games on xbox/ps3 have mp3 by default
Games on PC are flac

**Comment by 90Ninety** [comment]
Hmm 

> HDMI supports flac 5.1

So it should be currently possible for my NVIDIA card ( with its realtek chip I guess?) to output a FLAC 5.1 stream via HDMI? 



> HDMI ARC doesn't support flac 5.1 cause of bandwidth. It can only do mp3 5.1

Presuming my PC may be connected to an ARC port ; could this be the reason  why I am only getting 2 Channel to the receiver?   Or are there other issues here 

I just wanna be able to kick back and enjoy true 5.1 sound from my PC , without having to tear down the wiring , and re-wiring when finished  . Ideally I could stream straight from the On board ANdroid OS , where the sound would be sent ( 5.1) to the Home Theater via Optical  


Technically It is possible to wire the PC straight to the Home Theatre and get DDL /DD/DTS  but this means tearing down existing wiring . The TV is currently connected to the Home Theatre via Toshlink Optical , which works fine for the native TV Applications ( via Android )  and I use it this way . If I want to game and get 5.1 I would have to rewire the toshlink to the PC .  

My Current Wiring PC via HDMI>SONY Android TV>Hometheatre (Optical )

**Comment by vrokolos** [comment]
Optical can't do HDMI 5.1 raw
that's your problem right there.

What you need to do is PC > Hometheatre(HDMI) > SONY (HDMI)

This isn't moonlight related though :)

**Comment by 90Ninety** [comment]
What about non ARC ports , my TV has three other ports , only one is ARC

**Comment by vrokolos** [comment]
Output from TV using HDMI can only be done with ARC or optical which neither supports 5.1 pcm/raw audio so you're out of luck.

It is pretty straight forward. You just connected it all wrong :)

**Comment by 90Ninety** [comment]
> It is pretty straight forward. You just connected it all wrong :)

We will have to agree to disagree on this , everything else works fine . 
Playstation> via HDMI>TV > via Optical> Sound System = 5.1 DD/ DTS 
Android Native Video app>  >via Optical > Sound System = 5.1 DD / DTS 
DVD player> via HDMI>TV> via Optical > SOund system = 5.1/ DTS 
PC> via HDMI>TV> via Optical > sound system = FAIL 

I would say that there is lack of consistency or standards  in connectivity of devices . The common denominator of this problem , is the PC connecting to the TV , there is a problem between these devices . I am not replacing my expensive and otherwise fine working equipment .

**Comment by vrokolos** [comment]
The PC is the only one of those devices that outputs 5.1 PCM

**Comment by vrokolos** [comment]
https://forums.geforce.com/default/topic/956641/?comment=4948752
